contrib/mesos/pkg/executor/executor.go

```diff
@@ -257,38 +243,26 @@ func (k *KubernetesExecutor) Reregistered(driver bindings.ExecutorDriver, slaveI
 		}
 	}
 
-	k.initialRegistration.Do(k.onInitialRegistration)
-}
-
-func (k *KubernetesExecutor) onInitialRegistration() {
-	defer close(k.initialRegComplete)
-
 	// emit an empty update to allow the mesos "source" to be marked as seen
+	k.lock.Lock()
+	defer k.lock.Unlock()
+	if k.isDone() {
+		return
+	}
 	k.updateChan <- kubelet.PodUpdate{
-		Pods:   []*api.Pod{},
-		Op:     kubelet.SET,
-		Source: k.sourcename,
+		Pods: []*api.Pod{},
```

Comment:
we probably shouldn't be sending this during re-registration. the first time an executor boots, yes. on slave fail-over? no.